ArduCopter/mode_circle.cpp

```diff
@@ -39,6 +39,56 @@ void ModeCircle::run()
         pilot_yaw_override = true;
     }
 
+    // pilot changes to circle rate and radius
+    // skip if in radio failsafe
+    if (!copter.failsafe.radio && !is_zero(copter.circle_nav->get_control())) {
+        // update the circle controller's radius target based on pilot pitch stick inputs
+        float radius_current = copter.circle_nav->get_radius();         // circle controller's radius target, which begins as the circle_radius parameter
+        float pitch_stick = channel_pitch->norm_input_dz();             // pitch stick normalized -1 to 1
+        float nav_speed = copter.wp_nav->get_default_speed_xy();        // copter loiter speed
+        float radius_pilot_change = (pitch_stick * nav_speed) * G_Dt;   // rate of change
+        float radius_new = MAX(radius_current + radius_pilot_change,0); // new radius target
+
+        if (!is_equal(radius_current,radius_new)) {
+            copter.circle_nav->set_radius(radius_new);
+        }
+
+        // update the orbicular rate target based on pilot roll stick inputs
+        // skip if using CH6 tuning knob for circle rate
+        if (g.radio_tuning != TUNING_CIRCLE_RATE) {
+            float rate = copter.circle_nav->get_rate();                 // circle controller's rate target, which begins as the circle_rate parameter
+            float rate_current = copter.circle_nav->get_rate_current(); // actual rate, which is probably different from the target rate
+            float roll_stick = channel_roll->norm_input_dz();           // roll stick normalized -1 to 1
+            float rate_pilot_change = (roll_stick * G_Dt);              // rate of change
+            float rate_new = rate_current;                              // new rate target
+            static bool speed_changing = false;                         // true when the roll stick is being held to facilitate stopping at 0 rate
```

Comment:
it's probably best to move this static to mode.h so that it's a member of the ModeCircle class.

Reply:
Done